### PR TITLE
SIP-NN - Converters among optional Functions, PartialFunctions and extractor objects

### PR DESCRIPTION
**Pull request by Atry**
Discussed at https://contributors.scala-lang.org/t/converters-between-a-option-b-partialfunction-and-extractor-objects/287

### CODE REVIEW

**Comment by Atry** [comment]
I don't understand why build fails in CI.

Anyone can help?

**Comment by Philippus** [comment]
Yes, not related to your PR I think, but it's related to the tut upgrade we did, I've created a PR https://github.com/scala/docs.scala-lang/pull/1136

File: _sips/sips/2018-08-20-converters-among-optional-functions-partialfunctions-and-extractor-objects.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+## Implementation
+
+The idea was originally implemented in a library: [Extractor.scala](https://github.com/ThoughtWorksInc/Extractor.scala), which has been used in [Binding.scala](https://github.com/ThoughtWorksInc/Binding.scala/blob/10.0.x/XmlExtractor/src/main/scala/com/thoughtworks/binding/XmlExtractor.scala#L63) and [sbt-api-mappings](https://github.com/ThoughtWorksInc/sbt-api-mappings/blob/f4e1353/src/main/scala/com/thoughtworks/sbtApiMappings/ApiMappings.scala#L48).
+
+The new implementation aims to become part of core library. The pull request can be found at [#7111][2].
+
+## Drawbacks
+
+Why should we *not* do this. Be honest, these questions will come out during the
```
Comment (jvican) [inline _sips/sips/2018-08-20-converters-among-optional-functions-partialfunctions-and-extractor-objects.md]: I don't understand this section. Is this some kind of note for yourself? If not, can you clarify what you mean by this and list the drawbacks that you see from your side?

Comment (Atry) [reply _sips/sips/2018-08-20-converters-among-optional-functions-partialfunctions-and-extractor-objects.md]: It's a placeholder section copied from the SIP template. I just deleted the section now.

**Comment by Atry** [comment]
This SIP is not found in https://docs.scala-lang.org/sips/all.html. Is it the intended behavior?

**Comment by martijnhoekstra** [comment]
Note, https://github.com/scala/scala/pull/7111 indicates that this SIP may need to be skipped

**Pull request merged**